script/vm/global.lua

```diff
@@ -518,6 +519,33 @@ function vm.hasGlobalSets(suri, cate, name)
     return true
 end
 
+---@param src parser.object
+local function checkIsUndefinedGlobal(src)
+    local key = src[1]
+
+    local uri = guide.getUri(src)
+    local dglobals = util.arrayToHash(config.get(uri, 'Lua.diagnostics.globals'))
+    local rspecial = config.get(uri, 'Lua.runtime.special')
+
+    local node = src.node
+    return src.type == 'getglobal' and key and not (
+        dglobals[key] or
+        rspecial[key] or
+        node.tag ~= '_ENV' or
+        vm.hasGlobalSets(uri, 'variable', key)
+    )
+end
+
+---@param src parser.object
+---@return boolean
+function vm.isUndefinedGlobal(src)
+    local node = vm.compileNode(src)
+    if node._undefined_global == nil then
```

Comment:
My previous example used the wrong style. Please change the fields to lower camel case.